internal/error_helper.go

```diff
@@ -44,11 +46,11 @@ func IsNetworkError(err error) bool {
 // IsTimeoutError returns true if the observed error was a timeout error
 func IsTimeoutError(err error) bool {
 	for err != nil {
-		if strings.Contains(err.Error(), "Specified timeout reached") {
+		// See: https://apple.github.io/foundationdb/api-error-codes.html
+		// 1031: Operation aborted because the transaction timed out
+		if fdbError, ok := err.(fdb.Error); ok && fdbError.Code == 1031 {
```

Comment:
Our own Timeout error is probably the better choice otherwise we have to import the FDB client lib.

Reply:
Yes, I think that it's best to avoid having the FDB client library dependency in the internal package, because so many other things will depend on that. Maybe we could have the fdbclient package catch FDB timeout errors and wrap them in something that's easier to check for in other parts of the code.